drivers/flash/nrf_qspi_nor.c

```diff
@@ -319,14 +316,14 @@ static int qspi_erase(struct device *dev, u32_t addr, u32_t size)
 		return -EINVAL;
 	}
 
-	int rv = -EIO;
+	int rv = 0;
 	const struct qspi_nor_config *params = dev->config_info;
 
-	while (size) {
+	qspi_lock(dev);
+	while ((rv == 0) && (size > 0)) {
```

Comment:
Shouldn't we check if `size` is multiply of full sectors? Otherwise it seems that loop will erase everything it can in sectors/blocks  and then  return with error for the rest of requested range if it has not been fully aligned.

Reply:
Probably, but that should be done in another PR as this is the same loop used in other drivers.